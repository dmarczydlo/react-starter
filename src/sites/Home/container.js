import Site from './home';
import { connect } from 'react-redux';
import { updateValue } from '../../actions/counter';

export const mapStateToProps = (state) => {
    return state;
};

export const mapDispatchToProps = (dispatch) => {
    return { updateValue: type => dispatch(updateValue(type)) };
};

export default connect(mapStateToProps, mapDispatchToProps)(Site);
