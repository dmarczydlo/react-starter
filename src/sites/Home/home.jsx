import React from 'react';
import PropTypes from 'prop-types';

const Home = ({ updateValue, counter: { value } }) => {
    return (
        <div className="counter">
            <button
                id="increment"
                onClick={() => updateValue('INCREMENT')}
                type="button"
            >
                {'+'}
            </button>
            <div
                className="value"
                data-test="value"
            >
                {value}
            </div>
            <button
                id="decrement"
                onClick={() => updateValue('DECREMENT')}
                type="button"
            >
                {'-'}
            </button>

        </div>

    );
};

Home.propTypes = {
    counter: PropTypes.shape({ value: PropTypes.number.isRequired }).isRequired,
    updateValue: PropTypes.func.isRequired
};

export default Home;
