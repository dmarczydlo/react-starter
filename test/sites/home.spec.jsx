import React from 'react';
import Home from '../../src/sites/home';

describe('home site', () => {
    let wrapper;

    it('should be rendered', () => {
        wrapper = global.shallow(<Home />);
        expect(wrapper).toMatchSnapshot();
    });
});
