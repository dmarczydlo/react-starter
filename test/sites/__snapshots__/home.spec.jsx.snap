// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home site should be rendered 1`] = `
<div
  className="test"
>
  <Link
    replace={false}
    to="/child/1"
  >
    child 1
  </Link>
  <Link
    replace={false}
    to="/child/2"
  >
    child 2
  </Link>
</div>
`;
